internal/planner/view.go

```diff
@@ -117,3 +107,142 @@ func (n *viewNode) Close() error {
 
 	return nil
 }
+
+func convertBetweenMaps(srcMap *core.DocumentMapping, dstMap *core.DocumentMapping, src core.Doc) core.Doc {
```

Comment:
suggestion: would be nice to have a comment for this function. From the name it's not clear why it's called "convert" although both source and destination have the same type.

Reply:
I mean, the first two params to this function are `srcMap` and `dstMap`, and the place it is called is heavily documented with the reason for calling it.